src/lib/rpmver-private.c

```diff
@@ -0,0 +1,217 @@
+// Temporary forked internal copy of:
+// https://github.com/rpm-software-management/rpm/blob/rpm-4.16.1.3/rpmio/rpmver.c
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+/* NOTE(lucab): compatibility changes to avoid touching the source */
+#include <glib.h>
+#define xmalloc(_size) g_malloc((_size))
```

Comment:
Looks like the behaviour between those two is slightly different for the 0 case. `xmalloc` bumps it up to 1, while `g_malloc` just returns `NULL`. Mehh... let's just copy in `xmalloc` too? But maybe instead of the `vmefail` bit, we just `g_assert`.

Reply:
Fair poiint. However there are exactly two `xmalloc` calls in this file and in both cases it can be mathematically proved that the argument is >= 1. May I just note this in an explicit comment instead?

Reply:
Ack, SGTM!